tests/test_consumer/test_extractors/test_extractors_column_lineage.py

```diff
@@ -0,0 +1,346 @@
+    )
+    dataset_column_relation_type = extract_dataset_column_relation_type(transformation)
+    assert dataset_column_relation_type == DatasetColumnRelationTypeDTO.UNKNOWN
+
+
+@pytest.mark.parametrize(
+    ("type", "subtype", "expected_type"),
+    (
+        (
+            "DIRECT",
+            "TRANSFORMATION",
+            DatasetColumnRelationTypeDTO.TRANSFORMATION_MASKING,
+        ),
+        ("DIRECT", "AGGREGATION", DatasetColumnRelationTypeDTO.AGGREGATION_MASKING),
+        ("INDIRECT", "JOIN", DatasetColumnRelationTypeDTO.UNKNOWN),
+    ),
+)
+def test_extractors_extract_dataset_column_relation_type_masking(
+    type,
+    subtype,
+    expected_type,
+):
+    transformation = OpenLineageColumnLineageDatasetFacetFieldTransformation(
+        type=type,
+        subtype=subtype,
+        masking=True,
+    )
+    dataset_column_relation_type = extract_dataset_column_relation_type(transformation)
+    assert dataset_column_relation_type == expected_type
+
+
+@pytest.mark.parametrize(
+    ("type", "subtype", "expected_type"),
+    (
+        ("DIRECT", "TRANSFORMATION", DatasetColumnRelationTypeDTO.TRANSFORMATION),
+        ("INDIRECT", "JOIN", DatasetColumnRelationTypeDTO.JOIN),
+        ("INDIRECT", "NEW_SUBTYPE", DatasetColumnRelationTypeDTO.UNKNOWN),
+    ),
+)
+def test_extractors_extract_dataset_column_relation_type_without_masking(
+    type,
+    subtype,
+    expected_type,
+):
+    transformation = OpenLineageColumnLineageDatasetFacetFieldTransformation(
+        type=type,
+        subtype=subtype,
+        masking=False,
+    )
+    dataset_column_relation_type = extract_dataset_column_relation_type(transformation)
+    assert dataset_column_relation_type == expected_type
+
+
+def test_extractors_extract_direct_column_lineage(
+    extracted_spark_operation,
+    extracted_hive_dataset,
+    extracted_hdfs_dataset,
+    output_event_with_one_to_two_direct_column_lineage,
+):
+    operation = extracted_spark_operation
+
+    column_lineage = extract_column_lineage(
+        operation,
+        output_event_with_one_to_two_direct_column_lineage,
+    )
+    assert column_lineage == [
+        ColumnLineageDTO(
+            operation=operation,
+            source_dataset=extracted_hive_dataset,
+            target_dataset=extracted_hdfs_dataset,
+            dataset_column_relations=[
+                DatasetColumnRelationDTO(
+                    type=DatasetColumnRelationTypeDTO.AGGREGATION,
+                    source_column="source_col_1",
+                    target_column="column_1",
+                    fingerprint=None,
+                ),
+                DatasetColumnRelationDTO(
+                    type=DatasetColumnRelationTypeDTO.TRANSFORMATION,
+                    source_column="source_col_2",
+                    target_column="column_1",
+                    fingerprint=None,
+                ),
+            ],
+        ),
+    ]
+
+
+def test_extractors_extract_indirect_column_lineage(
```

Comment:
Please add a test case with legacy indirect lineage example:
```
col1=[DIRECT col1, INDIRECT col2], col2=[DIRECT col1, INDIRECT col2]
->
direct=[col1=[DIRECT col1], col2=[DIRECT col2]], indirect=[]
```

Reply:
Added `test_extractors_extract_legacy_indirect_column_lineage`

Reply:
Please add a comment that indirect lineage is ignored in this case